go/client/logclient.go

```diff
@@ -109,168 +104,37 @@ type GetProofByHashResponse struct {
 // http://ct.googleapis.com/pilot
 // |hc| is the underlying client to be used for HTTP requests to the CT log.
 func New(uri string, hc *http.Client) *LogClient {
-	if hc == nil {
-		hc = new(http.Client)
+	logClient, err := jsonclient.New(uri, hc, "")
+	if err != nil {
+		panic("JSONClient creation failed: " + err.Error())
```

Comment:
This was mostly a 'not changing the existing API' thing -- `client.New` doesn't return an `error`, so the error information is lost regardless.

(It's currently also a 'can't happen in practice' thing, because all the failure cases in `jsonclient.New` are key-related, but it might not stay that way forever.)

So I thought a panic better than a `return nil` (and probably panic just afterwards).

(rejig done).